api.yml

```
@@ -185,6 +210,41 @@ paths:
           description: not supported by this implementation!
 components:
   schemas:
+    IssueCredentialRequest:
+      type: object
+      properties:
+        payload:
```

```suggestion
        credential:
```

We can be more specific here, this is not just a payload, it's a credential.

---

it has no proof... I have similar grievances with the use of `credential` and `presentation` all over chapi... what is the term for a vc that has no proof?

similarly for the case of a JWT, this payload will be embedded as a `vc` property in the payload of the JWT.... 

The object by itself is not verifiable, and is has not cryptographic protection... it is not a credential.

---

I expect that as we dig at this, we will end with something that is more like @peacekeeper 's original API... because the JWT VC Data Model has its own places to define issuer, sub, etc... https://www.w3.org/TR/vc-data-model/#jwt-encoding

```
For backward compatibility with JWT processors, the following JWT-registered claim names MUST be used instead of, or in addition to, their respective standard verifiable credential counterparts:
```

Seems like we need to make this ONLY in addition to.... or they payload input would need to be different to get the same kind of VC on the way out... the wording of this section of the VC Data Model makes me very sad... why are we letting so much optionality in here...

---

> what is the term for a vc that has no proof?

"Credential" :) ... if it has a proof it's a "Verifiable Credential" ... see the Verifiable Credentials spec: https://www.w3.org/TR/vc-data-model/#terminology

---

I stand corrected :)

---

Regarding the JWT encoding implications of this decision, lets avoid polluting this PR with discussion of JWT, as its not a requirement for this version, if you have comments of JWT encoding, lets aggregate them here: https://github.com/w3c-ccg/vc-issuer-http-api/issues/31

---

> Seems like we need to make this ONLY in addition to.... or the payload input would need to be different to get the same kind of VC on the way out... the wording of this section of the VC Data Model makes me very sad... why are we letting so much optionality in here...

... because the JWT people wanted it and wrote that section. That section makes me very sad too, but it's what compromise looks like. If we didn't include that section in the spec, the JOSE people were going to walk.

The good news, though, is that all the JWT stuff (save for like one or two items) can be auto-generated from the VC, which is why the input type is probably still going to be `credential` in the JWT case with maybe one or two things going in the `options` field.

---

resolving, because its outdated.